File: Python/SinglyLinkedList.py
""" 
	Write a python Program to create a singly linked-list, insert at the front and display the contents of the linked-list
	Author: Mukul Milind Mishra
	Last Modified: 07-07-2019 
"""

class Node(object):

    def __init__(self, data=None, next_node=None):
        self.data = data
        self.next_node = next_node

    def get_data(self):
        return self.data

    def get_next(self):
        return self.next_node

    def set_next(self, new_next):
        self.next_node = new_next

class LinkedList(object):
	
	# The first architectural piece of the linked list is the ‘head node’, which is simply the top node in the list. 
	# When the list is first initialized it has no nodes, so the head is set to None.
	def __init__(self, head=None):
		self.head = head


	# This insert method takes data, initializes a new node with the given data, and adds it to the list. 
	# Technically you can insert a node anywhere in the list, but the simplest way to do it is to place it at the head of the list and point the new node at the old head
	def insert(self, data):
		new_node = Node(data)
		new_node.set_next(self.head)
		self.head=new_node

	def size(self):
		current = self.head
		count = 0
		while current:
			count += 1
			current = current.get_next()
		return count

	def search(self, data): 
		current = self.head
		found = False
		while current and Found is False:
			if current.get_data() == data:
				found = True
			else:
				current = current.get_next()
		if Current is None:
			raise ValueError("Data is not present in the linked-list")

		return Current

File: Java/LinkedList/SinglyLinkedList.java
package linkedlist;

class Node {
    private int data;
    private Node next;

    public Node(int data) {
        this.data = data;
        this.next = null;
    }

    public int getData() {
        return data;
    }

    public void setData(int data) {
        this.data = data;
    }

    public Node getNext() {
        return next;
    }

    public void setNext(Node next) {
        this.next = next;
    }

    // @Override
    // public String toString() {
    //     return "Node [data=" + data + "]";
    // }
    
}
class LinkedList {
    private Node head;
    private int size;

    public boolean isEmpty() {
        return head == null;
    }
    public int getSize() {
        return size;
    }

    public void addToFront(int data) {
        Node node = new Node(data);
        node.setNext(head);
        head = node;
        size++;
    }
    public Node removeFromFront() {
        if (isEmpty()) return null;
        size--;
        Node removedNode = head;
        head = head.getNext();
        removedNode.setNext(null);
        return removedNode;
    }
    public void printList() {
        Node current = head;
        System.out.print("HEAD -> ");
        while (current != null) {
            System.out.print(current.getData());
            System.out.print(" -> ");
            current = current.getNext();
        }
        System.out.println(" null ");
    }
}

public class SinglyLinkedList {
    public static void main(String [] args) {
        LinkedList newList = new LinkedList();
        newList.addToFront(1);
        newList.addToFront(2);
        newList.addToFront(3);
        newList.addToFront(4);

        newList.printList();
        System.out.println("Current Size of list is: " + newList.getSize());

        System.out.println("Removed the element from the front: " + newList.removeFromFront().getData());

        newList.printList();
        System.out.println("Current Size of list is: " + newList.getSize());

    }
}